pkg/nsx/services/ipaddressallocation/ipaddressallocation.go

```diff
@@ -127,6 +130,73 @@ func (service *IPAddressAllocationService) CreateOrUpdateIPAddressAllocation(obj
 	return true, nil
 }
 
+// CreateIPAddressAllocationIfNeeded is only for the restore of external address binding.
+func (service *IPAddressAllocationService) CreateIPAddressAllocationIfNeeded(addressBinding *v1alpha1.AddressBinding, subnetPort *v1alpha1.SubnetPort, restoreMode bool) error {
+	if !restoreMode {
+		// Only create the NSX IPAddressAllocation in restore stage
+		return nil
+	}
+	if len(addressBinding.Status.IPAddress) == 0 {
+		return nil
+	}
+	existingIPAddressAllocation, err := service.GetIPAddressAllocationByOwner(addressBinding)
+	if err != nil {
+		log.Error(err, "Failed to get IPAddressAllocation for AddressBinding", "AddressBinding", addressBinding)
+		return err
+	}
+	if existingIPAddressAllocation != nil {
+		log.V(1).Info("The IPAddressAllocation has been created, skipping", "AddressBinding", addressBinding)
+		return nil
+	}
+	nsxIPAddressAllocation, err := service.BuildIPAddressAllocation(addressBinding, subnetPort, restoreMode)
+	if err != nil {
+		return err
+	}
+	if nsxIPAddressAllocation == nil {
+		return fmt.Errorf("failed to build NSX IPAddressAllocation for AddressBinding %s", addressBinding.UID)
+	}
```

Comment:
For curiosity, is there any case error is not nil while nsxIPAddressAllocation is nil?

Reply:
I checked the code and found ant the nsxIPAddressAllocation won't be nil. Will delete this error output, thanks.